File: gamao/lua/dado.lua
m=require("modulo")

-- Códigos de retorno possíveis para o arcabouço
--ret= {{"Nome", 'Descrição'}}
ret = {{"OK", "Rodou sem erros"}}

-- param = {"Nome", "Tipo", "Descrição", Oculto do teste (default false)}
params = {{"pDado1", "int*", "Resultado do dado 1.\nEste parâmetro é passado por referência.", true},
          {"pDado2", "int*", "Resultado do dado 2.\nEste parâmetro é passado por referência.", true}}
--func {'Nome da função', 'Descrição',                                  Retornos,  Parâmetros, Nome Teste}
func = {{"Jogar Dados",    "Gera dois números aleatórios entre 1 e 6.", {"OK"},    params,     "jogar"}}

autores = {"Gabriel Barbosa de Oliveira", "Guilherme de Azevedo Pereira Marques", "Thiago Duarte Naves"}

--criar_modulo(nome,   id,    testes, mult_instâncias, cond_ret, funcoes, autores, arq_code,   arq_head,   arq_test,     arq_script)
m.criar_modulo("dado", "DAD", true,   false,           ret,      func,    autores, "dado.c",   "dado.h",   "testdado.c", "TesteDado.script")

File: gamao/dado.c
/**********************************************************************
*
*  $MCD Módulo de definição: Módulo dado
*
*  Arquivo gerado:              dado.c
*  Letras identificadoras:      DAD
*
*  Projeto: Disciplina INF 1301
*  Autores: gbo - Gabriel Barbosa de Oliveira
*           gapm - Guilherme de Azevedo Pereira Marques
*           tdn - Thiago Duarte Naves
*
*  $HA Histórico de evolução:
*     Versão  Autor             Data       Observações
*       1.00  gbo, gapm, tdn    18/09/2015 Início do desenvolvimento
*
*  $ED Descrição do módulo
*     Gera dois números aleatórios.
*
***********************************************************************/

#include   <stdio.h>
#include   <stdlib.h>
#include   "dado.h"
#include   <time.h>
#include   <assert.h>


/*****  Código das funções exportadas pelo módulo  *****/


/***************************************************************************
*
*  Função: DAD Jogar Dados
*  ****/

   DAD_tpCondRet DAD_JogarDados( int* pDado1 , int* pDado2 )
   {

      assert ( pDado1 != NULL && pDado2 != NULL ) ;

      * pDado1 = ( rand ( ) % 6 ) +1 ;
      * pDado2 = ( rand ( ) % 6 ) +1 ;

      return DAD_CondRetOK ;

   } /* Fim função: DAD Jogar Dados */


/*****  Código das funções encapsuladas pelo módulo  *****/


/********** Fim do módulo de implementação: Módulo dado **********/

